src/Corax/IndexWriter.cs

```diff
@@ -125,27 +146,82 @@ public long GetNumberOfEntries()
+                //       that will ensure faster turnaround and more efficient execution. 
                 var iterator = entryReader.ReadMany(tokenField);
                 while (iterator.ReadNext())
-                {
+                {                    
+                    // Because of how we store the data, either this is a sequence or a tuple, which also contains a sequence. 
                     var value = iterator.Sequence;
 
-                    using var _ = Slice.From(context, value, ByteStringType.Mutable, out var slice);
+                    var words = new Span<byte>(tempWordsSpace, bufferSize);
+                    var tokens = new Span<Token>(tempTokenSpace, tokenSize);
+                    _analyzer.Execute(value, ref words, ref tokens);
+
+                    for (int i = 0; i < tokens.Length; i++)
+                    {
+                        ref var token = ref tokens[i];
+
+                        using var _ = Slice.From(context, words.Slice(token.Offset, (int)token.Length), ByteStringType.Mutable, out var slice);
+                        if (field.TryGetValue(slice, out var term) == false)
+                        {
+                            var fieldName = slice.Clone(context);
+                            field[fieldName] = term = new List<long>();
+                        }
+
+                        AddMaybeAvoidDuplicate(term, entryId);
+                    }
+                }
+            }           
+            else if (fieldType.HasFlag(IndexEntryFieldType.Tuple) || fieldType.HasFlag(IndexEntryFieldType.Invalid) == false)
+            {
+                entryReader.Read(tokenField, out var value);
+
+                var words = new Span<byte>(tempWordsSpace, bufferSize);
+                var tokens = new Span<Token>(tempTokenSpace, tokenSize);
+                _analyzer.Execute(value, ref words, ref tokens);
+
+                for (int i = 0; i < tokens.Length; i++)
+                {
+                    ref var token = ref tokens[i];
+
+                    using var _ = Slice.From(context, words.Slice(token.Offset, (int)token.Length), ByteStringType.Mutable, out var slice);
                     if (field.TryGetValue(slice, out var term) == false)
                     {
                         var fieldName = slice.Clone(context);
                         field[fieldName] = term = new List<long>();
                     }
 
-                    AddMaybeAvoidDuplicate(term);
+                    AddMaybeAvoidDuplicate(term, entryId);
                 }
             }
-            else if (fieldType.HasFlag(IndexEntryFieldType.List))
+        }        
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        void AddMaybeAvoidDuplicate(List<long> term, long entryId)
+        {
+            // TODO: Do we want to index nulls? If so, how do we do that?
```

Comment:
`null` and `empty` strings should probably be a dedicated value.

Reply:
Yes, lets talk later how to do that.

Reply:
@redknightlois Maybe open a dedicated issue for that, so we won't forget about it?

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-17602